pkg/registry/utils.go

```diff
@@ -0,0 +1,41 @@
+package registry
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/url"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	urlfetcher "github.com/vmware/vic/pkg/fetcher"
+)
+
+// Reachable test if a registry is a valid registry for VIC use and returns a url with scheme prepended
+func Reachable(registry, schema, username, password string, registryCAs *x509.CertPool, timeout time.Duration, skipVerify bool) (string, error) {
```

Comment:
I wanted this function to be as basic as possible.  We can create another function that does what you're suggesting.  Currently only two functions in the entire code base calls Reachable so it's below my threshold for refactoring the higher level logic you're describing.